src/controllers/PartialUserFormController.php

```diff
@@ -113,4 +120,45 @@ protected function createOrUpdateSubmission($formData)
         // Return the ParentID to link the PartialSubmission to it's proper thingy
         return $editableField;
     }
+
+    /**
+     * Partial form
+     *
+     * @param HTTPRequest $request
+     * @return \SilverStripe\ORM\FieldType\DBHTMLText|void
+     * @throws \SilverStripe\Control\HTTPResponse_Exception
+     */
+    public function partial(HTTPRequest $request)
+    {
+        // Ensure this URL doesn't get picked up by HTTP caches
+        HTTPCacheControlMiddleware::singleton()->disableCache();
+
+        $key = $request->param('Key');
+        $token = $request->param('Token');
+
+        $partial = PartialFormSubmission::get()->find('Token', $token);
+        if (!$partial || !$partial->UserDefinedFormID) {
+            return $this->httpError(404);
+        }
+
+        if ($partial->generateKey($token) === $key) {
+
+            // TODO: Recognize visitor with the password
+            // TODO: Populate form values
+
+            $record = UserDefinedForm::get()->byID($partial->UserDefinedFormID);
```

Comment:
It might not be a UserDefinedForm in the case of an Elemental element :/ Do you have any thoughts on how to resolve that?